core/amber/src/main/scala/edu/uci/ics/texera/workflow/operators/regex/RegexOpExec.scala

```diff
@@ -6,12 +6,17 @@ import edu.uci.ics.texera.workflow.common.operators.filter.FilterOpExec
 import edu.uci.ics.texera.workflow.common.tuple.Tuple
 
 class RegexOpExec(val opDesc: RegexOpDesc) extends FilterOpExec {
-  val pattern: Pattern = Pattern.compile(opDesc.regex)
+  val pattern: Pattern =
+    if (opDesc.caseInsensitive) Pattern.compile(opDesc.regex, Pattern.CASE_INSENSITIVE)
+    else Pattern.compile(opDesc.regex)
   this.setFilterFunc(this.matchRegex)
 
   def matchRegex(tuple: Tuple): Boolean = {
-    val tupleValue = tuple.getField(opDesc.attribute).toString
-    pattern.matcher(tupleValue).find
+    val tupleValue = Option[Any](tuple.getField(opDesc.attribute)).map(x => x.toString)
+    if (tupleValue.isEmpty)
+      true
```

Comment:
why are we returning true if the text doesn't exist?

Reply:
agree, it should return false since null doesn't match anything. I've changed to false